src/dev/bazel/jsts_transpiler.bzl

```diff
@@ -2,28 +2,57 @@
 
 load("@npm//@babel/cli:index.bzl", _babel = "babel")
 
-def jsts_transpiler(name, srcs, build_pkg_name, root_input_dir = "src", config_file = ".babelrc", additional_args = ["--quiet"], **kwargs):
+def jsts_transpiler(name, srcs, build_pkg_name, web = False, root_input_dir = "src", config_file = "", additional_args = ["--quiet"], **kwargs):
   """A macro around the autogenerated babel rule.
 
   Args:
     name: target name
     srcs: list of sources
+    build_pkg_name: package name into the build folder
+    web: setup the correct presets to consume the outputs in the browser, defaults to "False" and optimizes for node
     root_input_dir: defines the root input dir to transpile files from, defaults to "src"
-    config_file: transpiler config file, it defaults to a package local .babelrc
+    config_file: defines a transpiler config file and overrides the inline presets as well as the web flag, it defaults to not defined
     additional_args: Any additional extra arguments, defaults to --quiet
     **kwargs: the rest
   """
+
+  inline_presets = [
+    "--presets",
+  ]
+
+  if web:
+    inline_presets += [
+      "@kbn/babel-preset/webpack_preset",
+    ]
+  else:
+    inline_presets += [
+      "@kbn/babel-preset/node_preset",
+    ]
+
   args = [
     "./%s/%s" % (build_pkg_name, root_input_dir),
-    "--config-file",
-    "./%s/%s" % (build_pkg_name, config_file),
     "--out-dir",
     "$(@D)",
     "--extensions",
     ".ts,.tsx,.js",
-  ] + additional_args
+  ] + inline_presets
+
+  data = []
+
+  if config_file:
+    data += [config_file]
+    args += [
+      "--config-file",
+      "./%s/%s" % (build_pkg_name, config_file),
+    ]
+  else:
+    args += [
+      "--no-babelrc"
+    ]
```

Comment:
Seems like we can remove support for the `config_file` argument

Reply:
Don't you anticipate any situation where we might need to specify an override config file? If that is the case then I think we can remove the support for the `config_file` arg

Reply:
In situations where we might reach for a config file I think it would be preferable to add another option like `web=True` to customize the config in a way that other packages can utilize if necessary.